project.py

```diff
@@ -185,14 +204,118 @@ def create_trackbars(self):
         cv.createTrackbar('lo', 'result', self.lo, 255, self.trackbar_callback)
         cv.createTrackbar('hi', 'result', self.hi, 255, self.trackbar_callback)
 
+    def __hamming_distance__(self, hash1, hash2):
+        hash1_len = len(hash1) * len(hash1[0])
+        hash2_len = len(hash2) * len(hash2[0])
+
+        if(hash1_len != hash2_len):
+            return 0
+        distance = 0
+        i = 0
+        for row in hash1:
+            distance += sum(map(lambda x: 0 if x[0] == x[1] else 1,
+                                zip(row, hash2[i])))
+            i = i + 1
+        return distance
+
+    def __check_change_orb__(self, image):
+        """
+        Computes ORB fetures on last saved slide and given image.
+        Tries to match features of these images and calculates ratio
+        of matched features with all found features in last saved slide
+
+        Args:
+            image(numpy.ndarray): Image from which to get features
+
+        Returns:
+            float: Similararity between last saved image and given image
+        """
+        orb = cv.ORB()
+        kp1, ds1 = orb.detectAndCompute(self.last_saved_slide, None)
+        kp2, ds2 = orb.detectAndCompute(image, None)
+        bf = cv.BFMatcher(cv.NORM_HAMMING, crossCheck=True)
+        matches = bf.match(ds1, ds2)
+        return float(len(matches))/len(kp1)
+
+    def __compute_ahash__(self, image):
+        """
+        Computes aHash. Implemantation based on
+        http://www.hackerfactor.com/blog/index.php?/archives/432-Looks-Like-It.html
+
+        Args:
+            image(numpy.ndarray): Image from which to compute the hash
+
+        Returns:
+            numpy.ndarray: 2D binary array with computed aHash
+        """
+        gray = cv.cvtColor(image, cv.COLOR_BGR2GRAY)
+        resized = cv.resize(gray, (8, 8), interpolation=cv.INTER_AREA)
+        mean = cv.mean(resized)[0]
+        return resized > mean
```

Comment:
@Foxiy Sorry for the buzz, it seems `resized` and `mean` are both numpy arrays, this will then return a 2D array as you promised in `Returns:` section of the docstring.